controllers/apidefinition_controller.go

```diff
@@ -100,56 +104,28 @@ func (r *ApiDefinitionReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		}
 
 		if desired.Spec.APIID == "" {
-			desired.Spec.APIID = encodeNS(req.NamespacedName.String())
+			upstreamRequestStruct.Spec.APIID = encodeNS(req.NamespacedName.String())
 		}
 
 		if desired.Spec.OrgID == "" {
-			desired.Spec.OrgID = env.Org
+			upstreamRequestStruct.Spec.OrgID = env.Org
 		}
 
 		util.AddFinalizer(desired, keys.ApiDefFinalizerName)
 
 		// we support only one certificate secret name for mvp
 		if len(desired.Spec.CertificateSecretNames) != 0 {
 			certName := desired.Spec.CertificateSecretNames[0]
-			secret := v1.Secret{}
-
-			err := r.Get(ctx, types.NamespacedName{Name: certName, Namespace: namespacedName.Namespace}, &secret)
-			if err != nil {
-				log.Error(err, "requeueing because secret not found")
-
-				return err
-			}
-
-			pemCrtBytes, ok := secret.Data["tls.crt"]
-			if !ok {
-				log.Error(err, "requeueing because cert not found in secret")
-
+			tykCertID, err, shouldRequeue := r.checkSecretAndUpload(ctx, certName, namespacedName, log, &env)
```

Comment:
Why do we need bool here? Can't we check err value here?

Reply:
I have updated the PR with the required changes.